fixed double type checking in TechDraw per #20131

jonzirk76 opened the pull request.

jonzirk76 commented on src/Mod/TechDraw/App/DrawViewClip.cpp:
```diff
@@ -73,7 +73,7 @@ void DrawViewClip::addView(App::DocumentObject* docObj)
-    auto* view = dynamic_cast<DrawView*>(docObj);
+    auto* view = static_cast<DrawView*>(docObj);
 
     if (!view) {
```
I removed the if statement block on 78, do we keep this line of code?

79        auto* link = dynamic_cast<App::Link*>(docObj);

jonzirk76 replied:
This whole block of code should be removed?

78 if (!view) {
        auto* link = dynamic_cast<App::Link*>(docObj);
        if (!link) {
            return;
        }

        if (link) {
            view = dynamic_cast<DrawView*>(link->getLinkedObject());
            if (!view) {
                return;
            }
        }
    }

jonzirk76 replied:
This whole block of code should be removed?

Line 78:
`if (!view) {
        auto* link = dynamic_cast<App::Link*>(docObj);
        if (!link) {
            return;
        }

        if (link) {
            view = dynamic_cast<DrawView*>(link->getLinkedObject());
            if (!view) {
                return;
            }
        }
    }`

benj5378 replied:
Not the whole block, only the guard clause:
delete `if (!view) {` and `}`

chennes commented on src/Mod/TechDraw/App/DrawViewClip.cpp:
```diff
@@ -73,19 +73,17 @@ void DrawViewClip::addView(App::DocumentObject* docObj)
-            if (!view) {
-                return;
-            }
+    if (link) {
```
This `if` is now redundant.

benj5378 replied:
The whole App::Link thing might be useless. If DrawViewPart is derived from App::Link, we don't need to cast. If it is not, the cast will always fail. Unreachable code.

Though I suggest fixing this in a separate PR. Author is very new to C++

chennes reviewed (changes requested):
One small change, otherwise looks good. We'll squash on merge so don't worry about it for this PR, but in the future you'll want to know about `git commit --amend --no-edit` :).

3x380V commented:
@jonzirk76, @WandererFan was talking about dropping changes to files not the files themselves.

chennes commented:
@jonzirk76 it looked like you were fighting with git a bit here, so I cleaned things up for you. I did it by rebasing to main, dropping the extra commits, and using an "interactive rebase" to squash the commits that we wanted to keep. If you're on Discord we have a very active community of developers there who would be happy to help out with things like this. git has a bit of a learning curve :).

jonzirk76 commented:
> @jonzirk76 it looked like you were fighting with git a bit here, so I cleaned things up for you. I did it by rebasing to main, dropping the extra commits, and using an "interactive rebase" to squash the commits that we wanted to keep. If you're on Discord we have a very active community of developers there who would be happy to help out with things like this. git has a bit of a learning curve :).

Thanks, that's exactly what was happening, but I believe benj cleaned things up already in another PR so I moved on.  This was a good crash course though!

chennes commented:
Great, thanks @jonzirk76 and @benj5378 for the cleaned up PR. Closing this one.
